### PR TITLE
Adding sectioned up happiness metric + fixing healthHappiness

### PR DESCRIPTION
**Pull request by mmilkowski36**

### CODE REVIEW

**Review by hadiyachishti — COMMENTED** [review]
👍

File: sugarscape.py [hunk 1]
```diff
@@ -48,10 +48,10 @@ def __init__(self, configuration):
...
-                             "meanTradePrice": 0, "tradeVolume": 0, "maxWealth": 0, "minWealth": 0, "meanHappiness": 0, "meanAgeAtDeath": 0,
-                             "seed": self.seed, "totalWealthLost": 0, "totalMetabolismCost": 0, "agentReproduced": 0,
-                             "agentStarvationDeaths": 0, "agentDiseaseDeaths": 0, "environmentWealthCreated": 0, "agentWealthTotal": 0, "environmentWealthTotal": 0,
-                             "agentWealthCollected": 0, "agentWealthBurnRate": 0, "agentMeanTimeToLive": 0, "agentWealths": [],
+                             "meanTradePrice": 0, "tradeVolume": 0, "maxWealth": 0, "minWealth": 0, "meanHappiness": 0, "meanWealthHappiness": 0, "meanHealthHappiness": 0,
+                             "meanSocialHappiness": 0, "meanFamilyHappiness": 0, "meanConflictHappiness":0, "meanAgeAtDeath": 0, "seed": self.seed, "totalWealthLost": 0, 
+                             "totalMetabolismCost": 0, "agentReproduced": 0, "agentStarvationDeaths": 0, "agentDiseaseDeaths": 0, "environmentWealthCreated": 0,
```
Comment (nkremerh) [inline sugarscape.py:53]: Trailing whitespace.

File: sugarscape.py [hunk 2]
```diff
@@ -48,10 +48,10 @@ def __init__(self, configuration):
         self.end = False # Simulation end flag
         # TODO: Remove redundant metrics
         self.runtimeStats = {"timestep": 0, "population": 0, "meanMetabolism": 0, "meanMovement": 0, "meanVision": 0, "meanWealth": 0, "meanAge": 0, "giniCoefficient": 0,
-                             "meanTradePrice": 0, "tradeVolume": 0, "maxWealth": 0, "minWealth": 0, "meanHappiness": 0, "meanAgeAtDeath": 0,
-                             "seed": self.seed, "totalWealthLost": 0, "totalMetabolismCost": 0, "agentReproduced": 0,
-                             "agentStarvationDeaths": 0, "agentDiseaseDeaths": 0, "environmentWealthCreated": 0, "agentWealthTotal": 0, "environmentWealthTotal": 0,
-                             "agentWealthCollected": 0, "agentWealthBurnRate": 0, "agentMeanTimeToLive": 0, "agentWealths": [],
+                             "meanTradePrice": 0, "tradeVolume": 0, "maxWealth": 0, "minWealth": 0, "meanHappiness": 0, "meanWealthHappiness": 0, "meanHealthHappiness": 0,
+                             "meanSocialHappiness": 0, "meanFamilyHappiness": 0, "meanConflictHappiness":0, "meanAgeAtDeath": 0, "seed": self.seed, "totalWealthLost": 0,
```
Comment (nkremerh) [inline sugarscape.py:52]: meanConflictHappiness value should have space after colon.
Trailing whitespace at end of line.

File: agent.py [hunk 3]
```diff
@@ -789,7 +794,7 @@ def findConflictHappiness(self):
                 return 1
             else:
                 return -0.5
-        return 0
+        else: return 0
```
Comment (nkremerh) [inline agent.py:797]: Put return on next line.

File: agent.py [hunk 4]
```diff
@@ -851,8 +856,7 @@ def findHappiness(self):
             return self.findWealthHappiness() + self.findConflictHappiness() + self.findFamilyHappiness() + self.findSocialHappiness() + self.findHealthHappiness()
 
     def findHealthHappiness(self):
-        if self.isSick:
-            return -1
+        if self.isSick(): return -1
```
Comment (nkremerh) [inline agent.py:859]: Put return on next line.

File: agent.py [hunk 5]
```diff
@@ -851,8 +856,7 @@ def findHappiness(self):
...
         else: return 1
```
Comment (nkremerh) [inline agent.py:860]: Put return on next line.

File: agent.py [hunk 6]
```diff
@@ -1012,6 +1016,7 @@ def findWealthHappiness(self):
                 elif math.log(self.wealth * 0.01, self.cell.environment.sugarscape.runtimeStats["meanWealth"] )*5 < -1:
                     return -1
                 return (math.log(self.wealth * 0.01, self.cell.environment.sugarscape.runtimeStats["meanWealth"] )*5)
+        return self.wealthHappiness
```
Comment (nkremerh) [inline agent.py:1019]: Should return calculated value, not a data member.

**Review by nkremerh — CHANGES_REQUESTED** [review]

**Comment by nkremerh** [comment]
Only nitpicks left here. Good work!

**Comment by nkremerh** [comment]
Added patch on those nitpicks. Please test and let me know if everything still behaves as intended. Revert anything that I broke!

**Review by nkremerh — APPROVED** [review]

**Pull request merged**